File: NumeroPrimo.py
# Este programa valida si un número es primo o no
# Utiliza funciones/métodos, bucles y estructuras condicionales.


#Entrada: Un número entero positivo y entero
def esPrimo(numero): # Función que recibe un número y retorna
    if numero < 2:  # si el numero es menor a 2, no es primo
        return False
    # Proceso: Bucle que recorre los números desde 2 hasta el número ingresado,
    # si el número es divisible por algún número en ese rango, no es primo
    for i in range(2, numero):
        if numero % i == 0:
            return False
    #Salida: Si el número no es divisible por ningún número en el rango, es primo
    return True

# Función que valida si el número ingresado es un entero positivo y entero
def validarNumeroPrimo():
    # Entrada: Un número entero positivo y entero
    while True:
        try:
            numero = int(input("Por favor ingrese un número: "))
        #Proceso: Si el número es menor a 0, se pide un número positivo y si no es un número entero, se pide un número entero
            if numero >= 0:
                break
            else:
                print("Ingrese un número positivo")
        except ValueError:
            print("Ingrese un número entero")
    #Salida: Se imprime si el númeroS es primo o no
    if esPrimo(numero):
        print(f"{numero} es primo")
    else:
        print(f"{numero} no es primo")
#Llamada a la función
validarNumeroPrimo()

File: numeroPrimo.cpp
#include <iostream>

// Entrada del programa y validación si el número es primo o no
bool esPrimo(int numero)
{
    // Proceso: Si el número es menor a 2, no es primo
    if (numero < 2)
    {
        return false;
    }
    // Bucle que recorre los números desde 2 hasta el número ingresado,
    // si el número es divisible por algún número en ese rango, no es primo
    for (int i = 2; i < numero; i++)
    {
        if (numero % i == 0)
        {
            return false;
        }
    }
    // Salida: Si no se cumple ninguna de las condiciones anteriores, el número es primo
    return true;
}

// Función que valida si el número ingresado es un entero positivo y entero
void validarNumeroPrimo()
{
    // Un número entero positivo y entero
    int numero;
    //Entrada: Se pide un número entero positivo
    while (true)
    {
        std::cout << "Por favor ingrese un número: ";
        std::cin >> numero;
        // Proceso: Si el número es positivo, se sale del bucle
        if (numero >= 0)
        {
            break;
        }
        else
        {
            std::cout << "Ingrese un número positivo" << std::endl;
        }
    }
    // Salida: Se imprime si el número es primo o no
    if (esPrimo(numero))
    {
        std::cout << numero << " es primo" << std::endl;
    }
    else
    {
        std::cout << numero << " no es primo" << std::endl;
    }
}

int main()
{
    // Llamada a la función
    validarNumeroPrimo();
    return 0;
}
